dart/lib/src/sentry_options.dart

```diff
@@ -459,6 +459,19 @@ class SentryOptions {
   /// ```
   Spotlight spotlight = Spotlight(enabled: false);
 
+  /// Configure a proxy to use for SDK API calls.
+  ///
+  /// On io platforms without native SDKs (dart, linux, windows), this will use
+  /// an 'IOClient' with inner 'HTTPClient' for http communication.
+  /// A http proxy will be set in returned for 'HttpClient.findProxy' in the
+  /// form 'PROXY <your_host>:<your_port>'.
+  /// When setting 'user' and 'pass', the 'HttpClient.addProxyCredentials'
+  /// method will be called wit empty 'realm'.
+  ///
+  /// On Android, the proxy settings are handled by the native SDK.
+  /// iOS and macOS native SDKs do not support proxy settings.
```

Comment:
Let's check how hard it is to set a proxy through the custom `urlSession` option

Reply:
I configured cocoa to set a `URLSession` on sentry options with the proxy setup. Tested with `mitmproxy` and it is recording the event when setting up `localhost:8080` as proxy. Running the example app as a macOS desktop app with both http and socks config tested.

```bash
mitmproxy --mode socks5 --listen-port 8080
```

```dart
options.proxy = SentryProxy(type: SentryProxyType.socks, host: "localhost", port: 8080);
```

<img width="2530" alt="Bildschirmfoto 2024-07-30 um 10 39 49" src="https://github.com/user-attachments/assets/10b05665-5b80-4f50-a734-876c5b90198c">